dashboard/src/actions/endpointAction.js

```diff
@@ -0,0 +1,25 @@
+import * as TYPES from "./types";
+
+export const fetchEndpoints = () => async (dispatch) => {
+  try {
+    const response = await fetch("/api/v1/endpoints");
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    const data = await response.json();
+    window.endpoints = data;
+    dispatch({
+      type: TYPES.SET_ENDPOINTS,
+      payload: { ...data },
+    });
```

Comment:
Perhaps orthogonal to Dave's question, I'll just note that what we are saving in the `window` object is a _reference_ to the endpoints object, while what we are passing to `dispatch()` is a _copy_ of it.  It seems like a reference should work in both instances.

Reply:
Yes, it works!